src/core/geometry/qgsbox3d.h

```diff
@@ -215,6 +215,20 @@ class CORE_EXPORT QgsBox3d
 
     bool operator==( const QgsBox3d &other ) const;
 
+    /**
+     * Scale the rectangle around a center QgsPoint.
+     *
+     * \since QGIS 3.22
+     */
+    void scale( double scaleFactor, const QgsPoint *c = nullptr );
```

Comment:
```suggestion
    void scale( double scaleFactor, const QgsPoint &center = QgsPoint() );
```

(You can then test for not set by checking whether center.isEmpty().

Reply:
I have just replicated the QgsRectangle::scale() behaviour. If you prefer, I can use your suggested solution.

Reply:
yes please!

Reply:
done